x-pack/test_serverless/api_integration/test_suites/common/saved_objects_management/relationships.ts

```diff
@@ -0,0 +1,473 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { schema } from '@kbn/config-schema';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { RoleCredentials } from '../../../../shared/services';
+
+export default function ({ getService }: FtrProviderContext) {
+  const svlCommonApi = getService('svlCommonApi');
+  const svlUserManager = getService('svlUserManager');
+  const supertest = getService('supertest');
+  const kibanaServer = getService('kibanaServer');
+  let roleAuthc: RoleCredentials;
+
+  const relationSchema = schema.object({
+    id: schema.string(),
+    type: schema.string(),
+    relationship: schema.oneOf([schema.literal('parent'), schema.literal('child')]),
+    meta: schema.object({
+      title: schema.string(),
+      icon: schema.string(),
+      editUrl: schema.maybe(schema.string()),
+      // visualizations don't declare an inAppUrl
+      inAppUrl: schema.maybe(
+        schema.object({
+          path: schema.string(),
+          uiCapabilitiesPath: schema.string(),
+        })
+      ),
+      namespaceType: schema.string(),
+      hiddenType: schema.boolean(),
+    }),
+  });
+  const invalidRelationSchema = schema.object({
+    id: schema.string(),
+    type: schema.string(),
+    relationship: schema.oneOf([schema.literal('parent'), schema.literal('child')]),
+    error: schema.string(),
+  });
+  const responseSchema = schema.object({
+    relations: schema.arrayOf(relationSchema),
+    invalidRelations: schema.arrayOf(invalidRelationSchema),
+  });
+
+  describe('relationships', () => {
+    before(async () => {
+      roleAuthc = await svlUserManager.createApiKeyForRole('admin');
+
+      await kibanaServer.savedObjects.cleanStandardList();
+      await kibanaServer.importExport.load(
+        'test/api_integration/fixtures/kbn_archiver/management/saved_objects/relationships.json'
+      );
+    });
+    after(async () => {
+      await kibanaServer.importExport.unload(
+        'test/api_integration/fixtures/kbn_archiver/management/saved_objects/relationships.json'
+      );
+      await kibanaServer.savedObjects.cleanStandardList();
+
+      await svlUserManager.invalidateApiKeyForRole(roleAuthc);
+    });
+
+    const baseApiUrl = `/api/kibana/management/saved_objects/relationships`;
+    const defaultTypes = ['visualization', 'index-pattern', 'search', 'dashboard'];
+
+    const relationshipsUrl = (type: string, id: string, types: string[] = defaultTypes) => {
+      const typesQuery = types.map((t) => `savedObjectTypes=${t}`).join('&');
+      return `${baseApiUrl}/${type}/${id}?${typesQuery}`;
+    };
+
+    describe('validate response schema', () => {
```

Comment:
I refactored this suite by assertions on search SO type to avoid deeply nested `describe` blocks that weren't executing `before/after` hooks.